pkg/provider/aws/auth/auth.go

```diff
@@ -95,16 +99,24 @@ func New(ctx context.Context, store esv1beta1.GenericStore, kube client.Client,
 	if prov.Region != "" {
 		config.WithRegion(prov.Region)
 	}
-	handlers := defaults.Handlers()
-	handlers.Build.PushBack(request.WithAppendUserAgent("external-secrets"))
-	sess, err := session.NewSessionWithOptions(session.Options{
-		Config:            *config,
-		Handlers:          handlers,
-		SharedConfigState: session.SharedConfigDisable,
-	})
-	if err != nil {
-		return nil, err
+
+	if savedSession != nil {
+		sess = savedSession
+	} else {
+
+		handlers := defaults.Handlers()
+		handlers.Build.PushBack(request.WithAppendUserAgent("external-secrets"))
+		sess, err = session.NewSessionWithOptions(session.Options{
+			Config:            *config,
+			Handlers:          handlers,
+			SharedConfigState: session.SharedConfigDisable,
+		})
+		if err != nil {
+			return nil, err
+		}
+		savedSession = sess
 	}
+
```

Comment:
If different SecretStores bound with different authentication methods and different permissions are used in a given context, won't effectively only make one of them to work (since then, the session will be saved globally?)

Reply:
Thanks @gusfcarvalho @moolen 
I have updated the PR, can you please take a look.

Also, I am not sure if makes sense to add the AWS session cache logic to `DefaultJWTProvider` function.

Reply:
Sure @moolen let me know if I can help anyhow

Reply:
@moolen I have update the PR for this feature to be a feature gate. Please check if PR looks good. Thanks.

Reply:
hey @albertollamaso, sorry i was on vacation the last dew days. With feature gate i meant to add a command line flag, e.g. `--experimental-enable-aws-session-cache` or similar.
I don't want to add that to the CRD, because this is (at least for now) a experimental thing and i don't want this to be covered by the deprecation policy. By prefixing the flag with `experimental` it should be obvious that this is a use-it-at-your-own-risk feature :)

Reply:
thanks @moolen for the clarification. I agreed that the feature gate implementation using `k8s/component-base` is out of scope of this PR. 

I have update the PR adding the feature gate the way I think you are suggesting. Please feel free to comment and flag any wrong implementation you see in the PR since I am still in the process of learning go.

I am pushing for this feature to be implemented, at least as experimental at the beginning since it will save a lot of API calls for large implementations of `ExternalSecrets` as we currently have in our insfrastructure.